File: CleanAndroid/app/src/main/java/de/phillippbertram/cleanandroid/internal/mvp/Presenter.java
package de.phillippbertram.cleanandroid.internal.mvp;

import android.support.annotation.Nullable;
import java.lang.ref.WeakReference;

/**
 * @author [PBE] Phillipp Bertram | phillipp.bertram@lmis.de
 */

public abstract class Presenter<V extends View> {


    /**
     * Die View für den verwendeten Presenter.
     */
    @Nullable
    private WeakReference<V> viewRef;


    public void bind(V view) {
        viewRef = new WeakReference<>(view);
    }

    public void unbind() {
        if (viewRef != null) {
            viewRef.clear();
            viewRef = null;
        }
    }


}

File: CleanAndroid/app/src/main/java/de/phillippbertram/cleanandroid/internal/mvp/Presenter.kt
package de.phillippbertram.cleanandroid.internal.mvp

import java.lang.ref.WeakReference

/**
 * @author [PBE] Phillipp Bertram | phillipp.bertram@lmis.de
 */

abstract class Presenter<in V : View> {

    /**
     * Die View für den verwendeten Presenter.
     */
    private var viewRef: WeakReference<V>? = null

    open fun bind(view: V) {
        viewRef = WeakReference(view)
    }

    open fun unbind() {
        if (viewRef != null) {
            viewRef!!.clear()
            viewRef = null
        }
    }


}
